plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -559,12 +559,12 @@ public Answer backupSnapshot(final CopyCommand cmd) {
                     physicalSize = Long.parseLong(tmp[1]);
                     finalPath = folder + File.separator + snapshotBackupUuid + ".vhd";
                 }
-
-                final String volumeUuid = snapshotTO.getVolume().getPath();
```

Comment:
This is where I was getting confused. I am inspecting the code in master, and we were only removing other snapshots, but the last one when it was not a full clone. You changed so we remove all but the last one always.

Reply:
Yes, some secondary storage (in our case Swift) cannot take partial snapshots, so we noticed the issue.